All_home_works/hw10/task1_get_result.py

```diff
@@ -0,0 +1,140 @@
+"""
+Ваша задача спарсить информацию о компаниях, находящихся в индексе S&P 500 с данного сайта:
+https://markets.businessinsider.com/index/components/s&p_500
+
+Для каждой компании собрать следующую информацию:
+
+    Текущая стоимость в рублях (конвертацию производить по текущему курсу, взятому с сайта центробанка РФ)
+    Код компании (справа от названия компании на странице компании)
+    P/E компании (информация находится справа от графика на странице компании)
+    Годовой рост/падение компании в процентах (основная таблица)
+    Высчитать какую прибыль принесли бы акции компании (в процентах),
+     если бы они были куплены на уровне 52 Week Low
+      и проданы на уровне 52 Week High (справа от графика на странице компании)
+
+Сохранить итоговую информацию в 4 JSON файла:
+
+    Топ 10 компаний с самими дорогими акциями в рублях.
+    Топ 10 компаний с самым низким показателем P/E.
+    Топ 10 компаний, которые показали самый высокий рост за последний год
+    Топ 10 комппаний, которые принесли бы наибольшую прибыль,
+     если бы были куплены на самом минимуме и проданы на самом максимуме за последний год.
+    Пример формата:
+
+    [
+    {
+     "code": "MMM",
+     "name": "3M CO.",
+     "price" | "P/E" | "growth" | "potential profit" : value,
+    },
+    ...
+    ]
+
+
+    P.S. по-максимуму использовать возможности параллелизма и асинхронности.
+
+    bs4
+    aiohttp
+
+"""
+import asyncio
+from multiprocessing import Pool
```

Comment:
What is the difference between async and multiprocessing?

Reply:
It's a huge difference. Asyncio works in one thread and can't work with a few functions at the same moment. Only one function works and when it starts sleeping or waiting, the control takes to another function.
And multiprocessing could work with few functions at the same moment in different processor cores, but they do not have a common memory.